bloom-filter/src/main/scala/bloomfilter/mutable/UnsafeBitArray.scala

```diff
@@ -66,5 +69,29 @@ class UnsafeBitArray(val numberOfBits: Long) {
     }
   }
 
+  @throws(classOf[java.io.ObjectStreamException])
+  /*private*/ def writeReplace : AnyRef = new UnsafeBitArray.SerializedForm(this)
+
+//  private def
+
   def dispose(): Unit = unsafe.freeMemory(ptr)
 }
+
+object UnsafeBitArray{
+  @SerialVersionUID(1l)
+  private class SerializedForm( @transient var unsafeBitArray: UnsafeBitArray ) extends Serializable{
```

Comment:
path of least resistamce, java serialization wasn't designed with imutability in mind.
taking the 'direct' approach would require a complete workout of this class, probably still leaving it in inconsistent state (uninitialized fields), and 'annotated' with vars where it currently have vals.